### ISSUE TITLE
Final phase still pointing to 20.3 repositories?

### ISSUE THREAD

**Issue by clefebvre** [linuxmint/mintupgrade#38]
> Running check ‘Final phase’
> Re-installing the meta-package
> DEBIAN_FRONTEND=noninteractive DEBIAN_PRIORITY=critical apt-get install –yes –no-install-recommends mint-meta-cinnamon
> Reading package lists…
> Building dependency tree…
> Reading state information…
> mint-meta-cinnamon is already the newest version (2021.11.18).
> The following packages were automatically installed and are no longer required:
> linux-headers-5.4.0-121 linux-headers-5.4.0-121-generic
> linux-image-5.4.0-121-generic linux-modules-5.4.0-121-generic
> linux-modules-extra-5.4.0-121-generic
> Use ‘sudo apt autoremove’ to remove them.
> 0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
> Re-installing the multimedia codecs
> DEBIAN_FRONTEND=noninteractive DEBIAN_PRIORITY=critical apt-get install –yes –no-install-recommends mint-meta-codecs
> Reading package lists…
> Building dependency tree…
> Reading state information…
> mint-meta-codecs is already the newest version (2021.11.18).
> The following packages were automatically installed and are no longer required:
> linux-headers-5.4.0-121 linux-headers-5.4.0-121-generic
> linux-image-5.4.0-121-generic linux-modules-5.4.0-121-generic
> linux-modules-extra-5.4.0-121-generic
> Use ‘sudo apt autoremove’ to remove them.
> 0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
> Installing new packages
> DEBIAN_FRONTEND=noninteractive DEBIAN_PRIORITY=critical apt-get install –yes –no-install-recommends neofetch ffmpegthumbnailer amd64-microcode intel-microcode celluloid drawing adwaita-icon-theme-full warpinator alsa-topology-conf alsa-ucm-conf mesa-vdpau-drivers mesa-vulkan-drivers cryptsetup-initramfs cryptsetup-run libreoffice-gtk3 gamemode gstreamer1.0-gtk3 gstreamer1.0-pipewire blueman sticky webp-pixbuf-loader xapp-appimage-thumbnailer xapp-epub-thumbnailer xapp-mp3-thumbnailer xapp-raw-thumbnailer
> Reading package lists…
> Building dependency tree…
> Reading state information…
> E: Unable to locate package webp-pixbuf-loader
> E: Unable to locate package xapp-appimage-thumbnailer
> E: Unable to locate package xapp-epub-thumbnailer
> E: Unable to locate package xapp-mp3-thumbnailer
> E: Unable to locate package xapp-raw-thumbnailer
> Error – Return code: 100

**Comment by clefebvre** [comment]
According to:

` mint-meta-codecs is already the newest version (2021.11.18).`

and:

```
E: Unable to locate package webp-pixbuf-loader
E: Unable to locate package xapp-appimage-thumbnailer
E: Unable to locate package xapp-epub-thumbnailer
E: Unable to locate package xapp-mp3-thumbnailer
E: Unable to locate package xapp-raw-thumbnailer
```

APT is not pointing towards Mint 21 which contains the thumbnailers and version 2022.07.03 of mint-meta-cinnamon.

**Comment by clefebvre** [comment]
This output shows the PostUpgradeCheck(). This is done some time after UpdateReposCheck() which job it is to point towards Mint 21.

We need the full output of the upgrade tool here to understand what happens during the previous phase.

Also, I'd like to know what the APT repositories point to, with the output of:

`inxi -r`

**Comment by cmacmackin** [comment]
I'm also experiencing this issue (which is strange, as I upgraded another laptop last night and didn't experience it then). The output when I ran `inxi -r` is

```
Repos:
  No active apt repos in: /etc/apt/sources.list 
  Active apt repos in: /etc/apt/sources.list.d/oem-somerville-charmander-meta.list 
  1: deb http://dell.archive.canonical.com/ jammy somerville
  2: deb http://dell.archive.canonical.com/ jammy somerville-charmander
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list 
  1: deb https://mirror.cov.ukservers.com/linuxmint una main upstream import backport
  2: deb http://archive.ubuntu.com/ubuntu focal main restricted universe multiverse
  3: deb http://archive.ubuntu.com/ubuntu focal-updates main restricted universe multiverse
  4: deb http://archive.ubuntu.com/ubuntu focal-backports main restricted universe multiverse
  5: deb http://security.ubuntu.com/ubuntu/ focal-security main restricted universe multiverse
  6: deb http://archive.canonical.com/ubuntu/ focal partner
```

**Comment by cmacmackin** [comment]
I've saved the full output of the upgrade procedure. Note that I tried changing the mirror I was pointing to a few times in initial efforts to debug this problem
[upgrade.log](https://github.com/linuxmint/mintupgrade/files/9301990/upgrade.log)
.

**Comment by clefebvre** [comment]
@cmacmackin looking at the log I see the following:

- The upgrade tool successfully made the OS point to jammy/vanessa
- During the download phase it encountered an error:

Err:2629 http://mirror.sov.uk.goscomb.net/ubuntu jammy/universe amd64 node-fs-readdir-recursive all 1.1.0-2
  403  Forbidden [IP: 77.75.110.242 80]

This is an error with the mirror or the state of the Ubuntu repositories at the time. If I try this URL manually now it works fine so I guess it was a temporary issue.

Either way, the tool tried twice and failed twice.

On its third try it succeeded but the repos had changed back to focal/una.

I may be wrong but I get the impression that you changed the repositories back to focal/una...? Maybe you used mintsources to do that? That would explain why the upgrade failed consequently.

Let me know if this is what happened and then we can think of making the tool detect that use case.

**Comment by liweiyuan** [comment]
the full output of the upgrade log 
[mintupgrade.log](https://github.com/linuxmint/mintupgrade/files/9330462/mintupgrade.log)

**Comment by cmacmackin** [comment]
@clefebvre 
> I may be wrong but I get the impression that you changed the repositories back to focal/una...? Maybe you used mintsources to do that? That would explain why the upgrade failed consequently.

I'm embarrassed to say that you are exactly right. I had only been thinking about the sources in terms of the base server URL and hadn't been considering that Mintsources would be pointing to the old version.

Incidentally, I went on to try updating the repositories by manually editing the config files. This got the installation to finish but it was badly broken by that point. When I rebooted I was presented with a TTY and the computer wasn't able to connect to the internet automatically. I couldn't work out how to connect it to the internet (possibly it was missing some packages necessary to do this). In the end I did a fresh reinstall. That computer had been formatted with home directories on a different partition than the system, so it all went very smoothly.
